### ISSUE TITLE
unnecessary_cast is unaffected by `// ignore: type=lint`

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/linter#4877]
Hello again!

Consider:

```dart
// ignore: type=lint
int value = 42 as int;
```

This will not silence the unnecessary_cast diagnostic.

I assume that this is another case of "the diagnostic is an _info_ instead of _lint_"

**Comment by bwilkerson** [comment]
Correct. `unnecessary_cast` is a warning.

**Comment by bwilkerson** [comment]
@srawlins

**Comment by rrousselGit** [comment]
It's an info for me :P
<img width="405" alt="Screenshot 2024-02-12 at 19 10 42" src="https://github.com/dart-lang/linter/assets/20165741/60696ac6-c5cb-4859-82da-3d568f794329">

**Comment by srawlins** [comment]
@rrousselGit you have set the severity to info. The "type" / "kind" is still lint.

Can I close or is there some specific feature you are looking for?

**Comment by srawlins** [comment]
Well, I guess it is a HintCode, `HintCode.UNNECESSARY_CAST`, exported as `WarningCode.UNNECESSARY_CAST`. https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/lib/src/error/codes.g.dart#L7264

**Comment by pq** [comment]
@srawlins: is this how it's meant to be or an oversight? Should it be migrated to a warning code?

**Comment by srawlins** [comment]
We could support `// ignore_for_file: type=warning`. I like that feature.

**Comment by bwilkerson** [comment]
> is this how it's meant to be or an oversight?

Removing the notion of hints is a work in process, and this is one of the not yet handled diagnostics.
